corehq/apps/reports/v2/filters/xpath_column.py

```diff
@@ -102,6 +108,50 @@ def format_value(self, value):
         return "'{}'".format(value) if value == '' else self._get_number(value)
 
 
+class DateXpathColumnFilter(BaseXpathColumnFilter):
+    title = ugettext_lazy("Date")
+    name = 'xpath_column_date'
+    data_type = DataType.DATE
+    choices = [
+        ChoiceMeta(ugettext_lazy("Date is"), 'date_is', '='),
+        ChoiceMeta(ugettext_lazy("Date is before"), 'date_before', '<'),
+        ChoiceMeta(ugettext_lazy("Date is after"), "date_after", '>'),
+    ]
+
+    @property
+    @memoized
+    def _timezone(self):
+        return get_report_timezone(self.request, self.domain)
+
+    def _adjust_to_utc(self, date):
+        localized = self._timezone.localize(date)
+        offset = localized.strftime("%z")
+        return date - datetime.timedelta(
+            hours=int(offset[0:3]),
+            minutes=int(offset[0] + offset[3:5])
+        )
+
+    def get_expression(self, property, choice_name, value):
+        if choice_name == 'date_is':
+            # This combined range has to be applied as using the '='
+            # operator will treat the value as a string match
```

Comment:
if I want to find everything on `2019-03-12`, the string it will compare it to might be `2019-03-12 00:00:00` or something thereabouts. In this case we can't search for the exact date because it would mean we must know the exact time.

Reply:
believe me. I tried this locally, and doing what I did above produced a result the user expects.

Reply:
I see. In your example you are doing the comparison on `last_modified` which is an HQ calculated property, not a case property.